internal/server/storage/s3/miniod/admin_client.go

```diff
@@ -0,0 +1,250 @@
+package miniod
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/google/uuid"
+
+	"github.com/lxc/incus/v6/shared/subprocess"
+)
+
+// AddServiceAccountResp is the response body of the add service account call.
+type AddServiceAccountResp struct {
+	AccessKey    string `json:"accessKey,omitempty"`
+	SecretKey    string `json:"secretKey,omitempty"`
+	SessionToken string `json:"sessionToken,omitempty"`
+}
+
+// InfoServiceAccountResp is the response body of the info service account call.
+type InfoServiceAccountResp struct {
+	AccountStatus string         `json:"accountStatus"`
+	ParentUser    string         `json:"parentUser"`
+	Policy        map[string]any `json:"policy"`
+}
+
+// AdminClient represents minio client.
+type AdminClient struct {
+	process *Process
+	alias   string
+	varPath string
+}
+
+// configDir returns path to the configuration directory for mc command.
+func (c *AdminClient) configDir() string {
+	return filepath.Join(c.varPath, "mc", "config")
+}
+
+// policyDir returns path to the location with temporary policy files.
+func (c *AdminClient) policyDir() string {
+	return filepath.Join(c.varPath, "mc", "policy")
+}
+
+// runClientCommand runs 'mc' command.
+func (c *AdminClient) runClientCommand(ctx context.Context, workDir string, extraArgs ...string) (string, error) {
+	args := []string{"-C", c.configDir()}
+
+	args = append(args, extraArgs...)
+
+	cmd := exec.CommandContext(ctx, "mc", args...)
+
+	if workDir != "" {
+		cmd.Dir = workDir
+	}
+
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		return stdout.String(), subprocess.NewRunError("mc", args, err, &stdout, &stderr)
```

Comment:
Since there could be people migrating from a version of Incus where the MinIO client library was used, it would be good to differentiate between no `mc` executable being found (ie, "Incus now depends on the MinIO Client; on most distributions you can install the `minio-client` package to fix this.") vs an error running the command itself.

Reply:
Fixed. We are returning error whenever `mc` nor `mcli` is not present on the system.